### PR TITLE
connect: Add `NextFunction` for `next`, fix `err` parameter

### PR DESCRIPTION
**Pull request by EvanHahn**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [code sample](https://gist.github.com/EvanHahn/e9dedbc01cadacc7295542bf47f7faed), [`next` documentation](https://github.com/senchalabs/connect#use-middleware)
- [x] ~Increase the version number in the header if appropriate.~
- [x] ~If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.~

### CODE REVIEW

File: types/connect/index.d.ts [hunk 1]
```diff
@@ -17,9 +18,11 @@ declare function createServer(): createServer.Server;
...
     export type ServerHandle = HandleFunction | http.Server;
 
+    type NextFunction = (err?: any) => void;
+
     export type SimpleHandleFunction = (req: http.IncomingMessage, res: http.ServerResponse) => void;
-    export type NextHandleFunction = (req: http.IncomingMessage, res: http.ServerResponse, next: Function) => void;
-    export type ErrorHandleFunction = (err: Error, req: http.IncomingMessage, res: http.ServerResponse, next: Function) => void;
+    export type NextHandleFunction = (req: http.IncomingMessage, res: http.ServerResponse, next: NextFunction) => void;
+    export type ErrorHandleFunction = (err: any, req: http.IncomingMessage, res: http.ServerResponse, next: NextFunction) => void;
     export type HandleFunction = SimpleHandleFunction | NextHandleFunction | ErrorHandleFunction;
```
Comment (geekley) [inline types/connect/index.d.ts]: @EvanHahn I think that `req` in the callbacks should use a subtype of `http.IncomingMessage` which also has `req.originalUrl: string`.
> In order to make middleware easier to write to be agnostic of the route, when the fn is invoked, the req.url will be altered to remove the route part (and the original will be available as req.originalUrl). For example, if fn is used at the route /foo, the request for /foo/bar will invoke fn with req.url === '/bar' and req.originalUrl === '/foo/bar'.
>-- https://www.npmjs.com/package/connect#appuseroute-fn

Comment (EvanHahn) [reply types/connect/index.d.ts]: That looks right to me. Would you be able to make a pull request? I don't think I'll have time, but could review it.

Comment (geekley) [reply types/connect/index.d.ts]: I've never done pull requests before... But I'll try to do it later.

Comment (EvanHahn) [reply types/connect/index.d.ts]: Feel free to email me at <me@evanhahn.com> if I can help!

**Pull request merged**